src/main/java/edu/ucsb/nceas/mdqengine/processor/AwardLookupCheck.java

```diff
@@ -51,84 +55,87 @@ public Result call() {
-					JSONObject json = (JSONObject) parser.parse(apiRetStr);
 
-					if (json == null) {
-						log.warn("No award information found for " + idString);
-						continue;
-					}
-					
-					log.debug("JSON=" + json.toJSONString());
+					// parse the JSON array directly
+					JSONParser parser = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE);
+					JSONObject json = (JSONObject) parser.parse(jsonStream);
 
 					JSONObject jsonResponse = (JSONObject) json.get("response");
 					JSONArray jsonAwards = (JSONArray) jsonResponse.get("award");
+
 					if (jsonAwards == null || jsonAwards.size() < 1) {
 						log.warn("No award information found for " + idString);
 						continue;
 					}
+
+					// Process each award in the JSON array
+
 					JSONObject jsonAward = (JSONObject) jsonAwards.get(0);
-				
-					// find the desired information about the award
+
+					// Extract desired information about the award
 					String title = (String) jsonAward.get("title");
 					String agency = (String) jsonAward.get("agency");
 					String id = (String) jsonAward.get("id");
 					String fundProgramName = (String) jsonAward.get("fundProgramName");
-					String primaryProgram = (String) jsonAward.get("primaryProgram");
-					
-					// add them all as outputs for search and grouping
+					JSONArray program = (JSONArray) jsonAward.get("primaryProgram");
+					String primaryProgram = (String) program.get(0);
+
+					// Add them all as outputs for search and grouping
 					outputs.add(new Output(title + " (" + agency + " " + id + ")"));
 					outputs.add(new Output(title));
 					outputs.add(new Output(agency));
 					outputs.add(new Output(id));
 					outputs.add(new Output(fundProgramName));
 					outputs.add(new Output(primaryProgram));
-					
-					// look up cross ref info as well
+
+					// Look up cross-reference info as well
 					outputs.addAll(this.lookupCrossRef(id));
 
-				} catch (Exception e) {
-					log.error("Could not look up award " + awardId + " at URL " + url.toString() + ": ", e);
+				} catch (MalformedURLException me) {
+					log.error("Could not look up award " + awardId + " at URL " + url.toString()
+							+ "malformed URL exception");
+					log.error(me.getMessage());
+					continue;
+				} catch (IOException ioe) {
+					log.error("Could not look up award " + awardId + " at URL " + url.toString() + "IO exception");
+					log.error(ioe.getMessage());
+					continue;
+				} catch (ParseException pe) {
+					log.error("Could not look up award " + awardId + " at URL " + url.toString() + "parse exception");
+					log.error(pe.getMessage());
 					continue;
 				}
 			}
-			
+
 			// set them all
 			result.setOutput(outputs);
-			
+
 		} else {
 			result.setStatus(Status.FAILURE);
 			result.setOutput(new Output("NA"));
 			log.warn("No award id given, cannot look-up funding");
 		}
-		
+
 		return result;
-		
+
 	}
-	
+
 	// TODO: implement this in a way that makes sense
 	private List<Output> lookupCrossRef(String awardId) {
```

Comment:
This method does a bunch of string manipulation, bet does not actually add any of those strings to the return value. It simply returns the empty ArrayList `outputs`.

If it's not needed, I'd recommend removing it completely (it's only called from 1 place)